### ISSUE TITLE
postcss 8.4.48 probably broke `stylelint-stylus`, `@stylistic/stylelint-plugin` and `stylelint-plugin-stylus` tests

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint-ecosystem-tester#62]
e.g. for `stylelint-plugin-stylus`

10/11 2:01AM: passes
10/11 10:19PM: 8.4.48 is released
11/11 1:58AM: fails

https://github.com/stylus/stylelint-stylus/blob/f57fc3379bad2c3eb8454f846ddec11bd609f00d/package.json#L78
https://github.com/postcss/postcss/releases/tag/8.4.48
https://www.npmjs.com/package/stylelint-plugin-stylus?activeTab=dependencies
https://github.com/stylelint/stylelint-ecosystem-tester/actions/workflows/test-package-stylelint-plugin-stylus-8e7.next.yml

related: #91

**Comment by ybiquitous** [comment]
I could reproduce an issue in the `@stylistic/stylelint-plugin` test suite with `postcss-html`:

```javascript
		{
			code: `<div>
<style>a {
  color: red;
}</style>


</div>`,
			message: messages.rejected,
			line: 4,
			column: 1,
		},
``` 

https://github.com/stylelint-stylistic/stylelint-stylistic/blob/32cff1c6107ca12bd149c81727fd6bc1377d3853/lib/rules/no-missing-end-of-source-newline/index.test.js#L95-L105

We can also see this issue on our demo site as follows.

With `postcss@8.4.47`, the lint problem's location is `3:2-3`. See [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIA8BnALgTwDYwHwEMACYAHQDsS1IsIAneQ2mKAbnMoF8kB6dbPciAA0IAGYBLHADl8AWziIYADzkAHHADoAFmllZh4CGQkBzBCFJlChEiHUBXE+LIpbDWwAE+WcenFheTBwfMjQAWgcnCmFya1swe3QIWQBlDFD8JTcbOwh0MBQUMJ09WyFYnNp7HFcEYgq4kC8g3zR-bjIIMNlfFGcTMJgyKDCIUTCUCHtaMBgwshgAdxCYbLQq1atCDnIOA0hjcRMAMTpZfDRzACtJsgNYVRRzS0bvGBC0bNsARgA2DQAHBoAExlCqebytdpvD4RLCOZxfEAAZg03zRYK2tlUeSohSRQIALBpCQB2TGNHH5QrFXRYJHo0kaAAMtl2IA4QA).

![Image](https://github.com/user-attachments/assets/7c99f834-d27d-4984-994a-27eedb8d318a)

With `postcss@8.4.48`, the lint problem's location is `2:9-10`. See [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIA8BnALgTwDYwHwEMACYAHQDsS1IsIAneQ2mKAbnMoF8kB6dbPciAA0IAGYBLHADl8AWziIYADzkAHHADoAFmllZh4CGQkBzBCFJlChEiHUBXE+LIpbDWwAE+WcenFheTBwfMjQAWgcnCmFya1swe3QIWQBlDFD8JTcbOwh0MBQUMJ09WyFYnNp7HFcEYgq4kC8g3zR-bjIIMNlfFGcTMJgyKDCIUTCUCHtaMBgwshgAdxCYbLQq1atCDnIOA0hjcRMAMTpZfDRzACtJsgNYVRRzS0bvGBC0bNsARgA2DQAHBoAExlCqebytdpvD4RLCOZxfEAAZg03zRYK2tlUeSohSRQIALBpCQDMY0cflCsVdFgkeiAOwaAAMtl2IA4QA).

![Image](https://github.com/user-attachments/assets/89f14c53-c699-44f6-88e6-14b3c046a1ca)

I suppose that this commit (https://github.com/postcss/postcss/commit/1a8b261404348c7ba495af0707d6d244d66cf566) for `rangeBy()` is related, but I'm not still sure which package we should fix, e.g. `postcss`, `postcss-html`, `@stylistic/stylelint-plugin`, or `stylelint`. 🤔 

@romainmenke If you have any ideas, please feel free to let us know. 🙏🏼

**Comment by Mouvedia** [comment]
> but I'm not still sure which package we should fix

I am pretty sure the ones that need to make the changes should be the plugins.

You could say that it should have been a minor version.
i.e. 8.4.47 → 8.5.0
But that's irrelevant now, if postcss consider it a fix it is a patch.

tl;dr: I have opened this issue so that we can track the issues on `@stylistic/stylelint-plugin` and `stylelint-plugin-styulus`

**Comment by romainmenke** [comment]
PostCSS is large enough that any change is a breaking change for someone.
This type of breakage was expected with those changes.

------

If I understand the issue correctly then this is roughly what is happening.

`postcss-html` (and likely similar packages) have `node.source.input.css` as the content of `<style>...</style>` **block** while also having `node.source.start` relative to the whole **document**.

We can't have both, those are contradictory.
This is best reported upstream in affected packages so that their respective communities can provide patches.

A minimal repro for `postcss-html`:

```js
const assert = require('assert');
const postcss = require("postcss");
const syntax = require("./lib/index.js")();
const document = syntax
	.parse(`<div>
<style>
a {
  color: red;
}</style>

</div>`, { syntax, from: 'foo' });

assert.deepStrictEqual(
	document.first.rangeBy({ index: document.first.source.input.css.length - 1 }),
	{ end: { column: 2, line: 5 }, start: { column: 1, line: 5 } }
);
```

I don't currently have much time and this is unlikely to improve over the next few months.
So any help with reporting this upstream would be greatly appreciated 🙇

**Comment by Mouvedia** [comment]
@firefoxic @ota-meshi

**Comment by ybiquitous** [comment]
Thank you for your help. Let me investigate it more.
Once I identify the cause, I'll open an issue upstream. 👍🏼

**Comment by romainmenke** [comment]
I've also filed an issue in PostCSS as I think we can make a small change there to make it easier for `postcss-html` and other to update their code:

https://github.com/postcss/postcss/issues/1995

**Comment by Mouvedia** [comment]
@ai @ajcwebdev
see postcss/postcss#1996

**Comment by ai** [comment]
I am going to release it today =^_^=

**Comment by Mouvedia** [comment]
@firefoxic we are down from 2 tests failing to 1.
i.e. https://github.com/stylelint/stylelint-ecosystem-tester/actions/runs/12758840683/job/35561584229 vs https://github.com/stylelint/stylelint-ecosystem-tester/actions/runs/12779252961/job/35623544702

**Comment by firefoxic** [comment]
It seems that this is no longer relevant. At least in relation to `@stylistic`…

For some reason, this issue was among my saved but done items 🤔
